### PR TITLE
feature: run releases barebone

### PR DESCRIPTION
**Pull request by InoMurko**
HELLO
## Overview

This should allow you to run child chain and watcher baremetal.

## Changes

- makefile ftw
## Testing
 lemme know if it works

so you want to try baremetal ehm?
First, I suggest you stop any containers you might be running (check if there are any with `docker ps`) and stop `plasma-deployer`, `geth`, `watcher`, `childchain`and `postgres`. Or just do 
1. `docker stop $(docker ps -q)`.
We then go all in.
Run:
2. `make start-services` this starts geth, postgres and plasma-deployer
Beware, plasma-deployer can be flaky. 
3. Open another terminal window an run
`make start-child_chain`
4. Open another terminal window an run
`make start-watcher`
5. Run `make get-alarms` once 3. and 4. are done booting, you should see curl responses from both services.
6. Explore the makefile with `make list`,
7. Other baremetal commands
```
make stop-child_chain
make stop-watcher
make remote-child_chain (allows you to attach yourself to the running node)
make remote-watcher (allows you to attach yourself to the running node)
```
To push code changes:
```
make update-child_chain
make update-watcher
```

Watcher and Child Chain need exported environment variables, they're found bin `bin/variables`.

Thanks @T-Dnzt !

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+	@echo "for Watcher and Child chain tagged with latest they will get pulled from our repository."
+	@echo "If you want to use your own image containers for Watcher and Child Chain"
+	@echo "use make docker-watcher && make docker-child_chain."
+	@echo "For rapid development that replaces containers with your code changes"
+	@echo "one can use make watcher-update or make child_chain-update."
+	@echo "BARE METAL DEVELOPMENT:"
+	@echo "ATTENTION ATTENTION ATTENTION"
+	@echo "-----------------------------"
+	@echo "You need to export environment variables."
+	@$(MAKE) -s raw-requirements
```
Comment (T-Dnzt) [inline Makefile]: Just tried running `make` after exporting the env variables, and I'm getting:

```
make[1]: *** No rule to make target `raw-requirements'.  Stop.
make: *** [help] Error 2
```

Is there something wrong with my `make`?

Comment (InoMurko) [reply Makefile]: oh, that rule got deleted! tnx

**Comment by InoMurko** [comment]
> I know this might seem pretty obvious in this case, but I think it's a good habit to take to have the actual steps to test in the PRs.
> 
> Here, we could have the following maybe:
> 
>     1. Make sure you've exported the environment variables (the script will remind you). Also curious why the script says`"You need to export environment variables."`, but it's actually part of the `raw-cluster` step: `set -e; . ./bin/variables`?
> 
>     2. Run `make`
> 
>     3. Enjoy your new setup, everything should work fine - just run a `mix test` to be sure
> 
> 
> What do you think @InoMurko?

What these rules do is that they standup the whole cluster, where child chain and watcher are spawned up *without* docker, but postgres, geth and plasma-deployer are still containerized. And because of all that... I don't know how I'm able to setup a circleci test + (which tests should I run, since we don't have end2end functional tests at hand).

I did a similar thing for `bin/setup` but in this case I'm out of ideas.

**Comment by T-Dnzt** [comment]
@InoMurko Oh no no, I just meant having those steps in the description of the PR :D No need to add any tests anywhere, it's just for us humans to try it out more easily (especially for people coming on the project now). Basically what I wrote could be copied and pasted in your PR desc :D

**Comment by InoMurko** [comment]
> @InoMurko Oh no no, I just meant having those steps in the description of the PR :D No need to add any tests anywhere, it's just for us humans to try it out more easily (especially for people coming on the project now). Basically what I wrote could be copied and pasted in your PR desc :D

pre-coffee symptoms!!! will do :)))

**Comment by T-Dnzt** [comment]
It's not working for me :'(

First crash:
```
Building Child Chain
init terminating in do_boot ({undef,[{Elixir.Distillery.Releases.Config.Provider,init,[[_]],[]},{init,eval_script,2,[]},{init,do_boot,3,[]}]})

Crash dump is being written to: erl_crash.dump...done
=CRASH REPORT==== 20-Sep-2019::15:21:35.717683 ===
  crasher:
    initial call: application_master:init/4
    pid: <0.76.0>
    registered_name: []
    exception exit: {bad_return,
                        {{elixir,start,[normal,[]]},
                         {'EXIT',
                             {undef,
                                 [{elixir,start,[normal,[]],[]},
                                  {application_master,start_it_old,4,
                                      [{file,"application_master.erl"},
                                       {line,277}]}]}}}}
      in function  application_master:init/4 (application_master.erl, line 138)
    ancestors: [<0.75.0>]
    message_queue_len: 1
    messages: [{'EXIT',<0.77.0>,normal}]
    links: [<0.75.0>,<0.43.0>]
    dictionary: []
    trap_exit: true
    status: running
    heap_size: 987
    stack_size: 27
    reductions: 197
  neighbours:

{"init terminating in do_boot",{undef,[{'Elixir.Distillery.Releases.Config.Provider',init,[[{'Elixir.OMG.Eth.ReleaseTasks.SetContract',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.Eth.ReleaseTasks.SetEthereumClient',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.DB.ReleaseTasks.SetKeyValueDB',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.WatcherRPC.ReleaseTasks.SetEndpoint',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.WatcherRPC.ReleaseTasks.SetTracer',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.Status.ReleaseTasks.SetSentry',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.Status.ReleaseTasks.SetTracer',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.Watcher.ReleaseTasks.SetChildChain',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.Watcher.ReleaseTasks.SetDB',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.Watcher.ReleaseTasks.SetTracer',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]}]],[]},{init,eval_script,2,[]},{init,do_boot,3,[]}]}}
Unable to configure release!
```

Then it runs for a while, doing cool stuff. And then, boom:

```
Init Child Chain DB
init terminating in do_boot ({undef,[{Elixir.Distillery.Releases.Config.Provider,init,[[_]],[]},{init,eval_script,2,[]},{init,do_boot,3,[]}]})

Crash dump is being written to: erl_crash.dump...done
=CRASH REPORT==== 20-Sep-2019::15:21:42.963074 ===
  crasher:
    initial call: application_master:init/4
    pid: <0.76.0>
    registered_name: []
    exception exit: {bad_return,
                        {{elixir,start,[normal,[]]},
                         {'EXIT',
                             {undef,
                                 [{elixir,start,[normal,[]],[]},
                                  {application_master,start_it_old,4,
                                      [{file,"application_master.erl"},
                                       {line,277}]}]}}}}
      in function  application_master:init/4 (application_master.erl, line 138)
    ancestors: [<0.75.0>]
    message_queue_len: 1
    messages: [{'EXIT',<0.77.0>,normal}]
    links: [<0.75.0>,<0.43.0>]
    dictionary: []
    trap_exit: true
    status: running
    heap_size: 987
    stack_size: 27
    reductions: 197
  neighbours:

{"init terminating in do_boot",{undef,[{'Elixir.Distillery.Releases.Config.Provider',init,[[{'Elixir.OMG.Eth.ReleaseTasks.SetContract',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.Eth.ReleaseTasks.SetEthereumClient',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.DB.ReleaseTasks.SetKeyValueDB',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.ChildChainRPC.ReleaseTasks.SetEndpoint',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.ChildChainRPC.ReleaseTasks.SetTracer',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.Status.ReleaseTasks.SetSentry',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]},{'Elixir.OMG.Status.ReleaseTasks.SetTracer',[<<"${RELEASE_ROOT_DIR}/config/config.exs">>]}]],[]},{init,eval_script,2,[]},{init,do_boot,3,[]}]}}
Unable to configure release!
```

Then it hangs :/

**Pull request merged**